hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -230,17 +235,8 @@ collectComTx networkId vk initialThreadOutput commits =
       headDatumAfter
   headDatumAfter =
     mkTxOutDatum Head.Open{Head.parties = initialParties, utxoHash, contestationPeriod = initialContestationPeriod}
-  -- NOTE: We hash tx outs in an order that is recoverable on-chain.
-  -- The simplest thing to do, is to make sure commit inputs are in the same
-  -- order as their corresponding committed utxo.
-  extractSerialisedTxOut d =
-    case fromData $ toPlutusData d of
-      Nothing -> error "SNAFU"
-      Just ((_, _, Just o) :: Commit.DatumType) -> Just o
-      _ -> Nothing
 
-  utxoHash =
-    Head.hashPreSerializedCommits $ mapMaybe (extractSerialisedTxOut . snd . snd) orderedCommits
+  utxoHash = toBuiltin $ hashTxOuts $ toList committed
```

Comment:
Revert this change and order the deserialised `SerialisedTxOut` value's `TxOutRef` field to order the outputs